### DISCUSSION TITLE
Option for context.AddOutValue() to not flag an object as dirty.

### DISCUSSION THREAD

**Discussion by lwmorris067** [MarimerLLC/csla#4247] [Ideas]
Let's say you have a business object with DateOfBirth and Age. After you fetch the business object from the db and load the DateOfBirth property, you do a BusinessRules.CheckRules to run the AgeRule. Well, now the business object is dirty. I have to manually add a MarkClean() before I leave the Fetch method.

It would be nice to have the equivalent of a SetProperty and LoadProperty within the business rule. Maybe context.SetOutValue and context.LoadOutValue.

I hope that all made sense.

**Comment by StefanOssendorf** [comment]
Hi. I think you want something like #4201 , right?
The only workaround I know to get a similiar result is:
You can do something like this within the rule itself
`LoadProperty(context.Target, YourNamePropertyInfo, "newValue");`

That will load the value immediatly but without any rule/property changed triggers

**Reply by lwmorris067** [reply]
It appears so. I don't care about the serialization/deserialization of certain fields, just an option for not marking the object diry.

**Reply by StefanOssendorf** [reply]
If you don't care about the serialization and such. What about a normal C# property?
